### PR TITLE
NAS-128209 / 24.10 / Implement slide in form for cloud backups

### PR DESCRIPTION
**Pull request by bvasilenko**
**Testing**

Test new slide-in form, please see ticket for details

### CODE REVIEW

**Review by AlexKarpov98 — COMMENTED** [review]
@bvasilenko - please add tests for the form component & pull changes from the master branch, there are updates related to the Cloud Backups.

File: src/app/pages/data-protection/cloud-backup/cloud-backup-form/cloud-backup-form.component.html [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        >
+          <ix-input
+            formControlName="description"
+            [label]="'Description' | translate"
+          ></ix-input>
+          <ix-input
+            formControlName="keep_last"
+            type="number"
+            [label]="'Keep Last' | translate"
+            [required]="true"
```
Comment (AlexKarpov98) [inline src/app/pages/data-protection/cloud-backup/cloud-backup-form/cloud-backup-form.component.html:60]: would be good to add help tooltips for the inputs.

File: src/app/pages/data-protection/cloud-backup/cloud-backup-form/cloud-backup-form.component.html [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+          <ix-scheduler
+            formControlName="schedule"
+            [required]="true"
+            [label]="'Schedule' | translate"
+          ></ix-scheduler>
+        </ix-fieldset>
+      </div>
+
+      <ix-form-actions>
+        <button mat-button type="button" matDialogClose ixTest="cancel">
```
Comment (AlexKarpov98) [inline src/app/pages/data-protection/cloud-backup/cloud-backup-form/cloud-backup-form.component.html:77]: I think we don't need this button, it's not working for me.
We don't have Cancel buttons on such similar forms.

File: src/app/pages/data-protection/cloud-backup/cloud-backup-form/cloud-backup-form.component.ts [hunk 3]
```diff
@@ -1,18 +1,290 @@
...
+        error: () => {
+          this.isLoading = false;
+          this.form.controls.bucket.disable();
+          this.form.controls.bucket_input.enable();
+          this.cdr.markForCheck();
+        },
+      });
+  }
+
+  getBucketsNodeProvider(): TreeNodeProvider {
```
Comment (AlexKarpov98) [inline src/app/pages/data-protection/cloud-backup/cloud-backup-form/cloud-backup-form.component.ts:156]: Would be good to fix this code duplication, you can add comment for that 

Screenshot 2024-04-12 at 16 30 46

**Comment by AlexKarpov98** [comment]
as well I can't submit the form:
Screenshot 2024-04-13 at 09 50 59

**Comment by AlexKarpov98** [comment]
one more issue I've found:
Screenshot 2024-04-13 at 09 57 14

**Comment by AlexKarpov98** [comment]
@bvasilenko - do we need to allow array here?
Screenshot 2024-04-13 at 09 58 32

I tried to save with array and got error.

Maybe we should do it like single element selection? `[multiple]="false"`
Screenshot 2024-04-13 at 09 58 41

**Comment by bvasilenko** [comment]
@AlexKarpov98 all the requests above should be addressed now

**Review by AlexKarpov98 — COMMENTED** [review]
@bvasilenko if I select 2 `Source path` items, I can't submit, it only works with 1 selected item.

https://github.com/truenas/webui/assets/22980553/88117683-29c8-4297-b9e9-07d78a6ef803

File: src/app/pages/data-protection/cloud-backup/cloud-backup-form/cloud-backup-form.component.html [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    <form class="ix-form-container" [formGroup]="form" (submit)="onSubmit()">
+      <div class="fieldsets">
+        <ix-fieldset
+          class="fieldset"
+          [title]="'Local' | translate"
+        >
+          <ix-explorer
+            formControlName="path"
+            [label]="'Source Path' | translate"
+            [tooltip]="'Select the directories or files to be sent to the cloud for backup.' | translate"
```
Comment (AlexKarpov98) [inline src/app/pages/data-protection/cloud-backup/cloud-backup-form/cloud-backup-form.component.html:13]: would be good to extract tooltips as we do for other forms
example `[tooltip]="helptext.ui_certificate.tooltip | translate"`

**Comment by bvasilenko** [comment]
> @bvasilenko if I select 2 Source path items, I can't submit, it only works with 1 selected item.

@AlexKarpov98 You can select multiple datasets within a pool:

https://github.com/truenas/webui/assets/20611516/78a9f0f7-9ae0-478a-87f8-a9d35211b672

**Review by AlexKarpov98 — APPROVED** [review]
Look good!
I think we can open form with `false` as the second option: 

 `this.chainedSlideInService.open(CloudBackupFormComponent, false, row)`

So it's takes less space

**Pull request merged**